htlcswitch/switch.go

```diff
@@ -494,7 +494,11 @@ func (s *Switch) forward(packet *htlcPacket) error {
 			}
 			addErr := ErrIncompleteForward
 
-			return s.failAddPacket(packet, failure, addErr)
+			linkError := NewDetailedLinkError(
+				failure, FailureDetailIncompleteForward,
+			)
+
+			return s.failAddPacket(packet, linkError, addErr)
```

Comment:
There is one case where we send in a nicely formatted user facing error about [not satisfying policy](https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/switch.go#L1070) for our own sends, but other than that it's duplicates.

So I think it's ok to just remove it? The only info that case is adding is the payment id, which is also logged. 

Instead of:
`incoming HTLC(hash) violated target outgoing link (id=pid) policy: unknown next lnk`

We'd just return:
`unknown next peer`

Reply:
I think it would be good to simplify this, especially if it just about where to log. For a regular rpc call to `SendPayment`, this won't be returned anyway. Also came across this is in https://github.com/lightningnetwork/lnd/pull/3915#discussion_r372296882

Reply:
> For a regular rpc call to SendPayment, this won't be returned anyway. 

Didn't realize this was the case, I thought that was the point of these. Removed in the following commit.

Reply:
For locally initiated payments, I don't think it even gets here. A bit further up a check `if packet.incomingChanID == hop.Source { return err}` is present